store/smt.go

```diff
@@ -246,15 +258,12 @@ func (s *SMT) traverse() (err lib.ErrorI) {
 		}
 		// assert current key isn't nil
 		if currentKey == nil {
-			panic("nil current key")
+			return ErrInvalidMerkleTree()
 		}
 		// load the bytes into the key
 		s.current.Key.fromBytes(currentKey)
```

Comment:
It appears as if this call is unnecessary as `s.current, err = s.getNode(currentKey)` does this

Reply:
You removed the wrong part. Revert

Reply:
Ensure you fully understand the review comment before making these changes